knotx-core/src/test/java/io/knotx/server/KnotxServerRoutingTest.java

```diff
@@ -143,20 +131,14 @@ public void whenRequestingPostGlobalPath_expectGlobalBC(
     createSimpleKnot(vertx, "B-engine", "+B", "go-c");
     createSimpleKnot(vertx, "C-engine", "+C", null);
 
-    testPostRequest(vertx, "/content/simple.html", resp -> {
+    testPostRequest(context, vertx, "/content/simple.html", resp -> {
       assertEquals(HttpResponseStatus.OK.code(), resp.statusCode());
       assertNotNull(resp.getHeader(EXPECTED_RESPONSE_HEADER));
       assertEquals(EXPECTED_XSERVER_HEADER_VALUE,
           resp.getHeader(EXPECTED_RESPONSE_HEADER));
-      resp.bodyHandler(body -> {
-        try {
-          assertEquals("global+B+C", body.toString(),
-              "Wrong engines processed request, expected " + "global+B+C");
-        } catch (Exception e) {
-          context.failNow(e);
-        }
-        context.completeNow();
-      });
+
+      assertEquals("global+B+C", resp.bodyAsString(),
+          "Wrong engines processed request, expected " + "global+B+C");
```

Comment:
This can be merged into a single string.

Reply:
fixed